examples_tests/13.BRDF_Explorer/BRDFExplorerApp.cpp

```diff
@@ -351,6 +357,14 @@ vec3 calculateSurfaceNormal(in vec3 position, in vec2 uv, in vec3 normal, in vec
 void main() {
     vec3 N = normalize(Normal);
     const vec2 texCoords = vec2(TexCoords.x, 1.0-TexCoords.y);
+    vec3 dp1 = dFdx( WorldPos );
+    vec3 dp2 = dFdy( WorldPos );
+    vec2 duv1 = dFdx( TexCoords );
+    vec2 duv2 = dFdy( TexCoords );
+    vec3 dp2perp = cross( dp2, N );
+    vec3 dp1perp = cross( N, dp1 );
+    vec3 T = normalize(dp2perp * duv1.x + dp1perp * duv2.x);
+    vec3 B = normalize(dp2perp * duv1.y + dp1perp * duv2.y);
```

Comment:
Because of texture warping on curved surfaces you can bet on TBN not being an orthogonal frame.
This will fuck up all the lighting computation that needs to know about tangents.

So maybe make `vec3 B = cross(N,T);` (don't normalize, its redundant)